### ISSUE TITLE
Question : the shape of matrix in svd

### ISSUE THREAD

**Issue by baker-Xie** [NicolasHug/Surprise#129]
#### Description
Dear Nicolas : 
In the SVD method, suppose we have a full_dataset :
1000 record
100 user
95 item
if we do train/test set split in the full_dataset , we may got two set that have different user and item:
train_set : 
800 record
90 user
80 item
test_set : 
100 record
15 user 
16 item
Then I have a question:

-----------------code-----------------
train_set, test_set = train_test_split(full_dataset, test_size = 0.2)
algo = svd(n_factors = 10)
algo.fit(train_set)               
algo.predict(test_set)      
In your code , you build a user_matrix = [90, n_factors]  and item_matrix = [ 80, n_factors] 
and the test_set may have user_id that's not in the train_set ,  I don't know how you predict the rating because I cannot see the source code in the method 'estimate' .

I also know that somebody else's code of SVD may build a  user_matrix = [100, n_factors]  and item_matrix = [ 95, n_factors] ,  which have different shape of your code. In this method, it's easy to understand how to predict the rating in test_set.
But, which method is correct?

Thanks
 

#### Steps/Code to Reproduce

None
#### Expected Results
None
#### Actual Results
None

**Comment by NicolasHug** [comment]
If your trainset has U users and I items, then the matrix sizes should be (U x F) and (I x F), as surprise does.
You can find the source code [here](https://github.com/NicolasHug/Surprise/blob/master/surprise/prediction_algorithms/matrix_factorization.pyx#L18)
